live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java

```diff
@@ -174,23 +201,69 @@ else if (value instanceof Integer)
         {
             return "integer";
         }
-        else if (value instanceof Float)
+        else if (value instanceof Float || value instanceof Double)
         {
             return "float";
         }
-        else if (value instanceof Collection collection)
+        else if (value instanceof Collection<?>)
         {
-            if (collection.isEmpty())
-            {
-                throw new IllegalArgumentException("Empty collections should not be passed to selectTypeByValue.");
-            }
-            return selectTypeByValue(collection.stream().findAny());
+            return "array";
+        }
+        else if (value instanceof Map<?, ?>)
+        {
+            return "object";
         }
         return "string";
     }
 
-    private String getLowerCase(Object object)
+    private void writeNestedProperties(JsonGenerator jgen, Map<?, ?> nestedMap) throws IOException
     {
-        return object.toString().toLowerCase(ENGLISH);
+        jgen.writeObjectFieldStart("value");
+
+        for (Map.Entry<?, ?> entry : nestedMap.entrySet())
+        {
+            String key = entry.getKey().toString();
+            Object nestedValue = entry.getValue();
+
+            jgen.writeObjectFieldStart(key);
+
+            String type = determineType(nestedValue);
+            jgen.writeStringField("type", type);
+
+            if ("object".equals(type))
+            {
+                writeNestedProperties(jgen, (Map<?, ?>) nestedValue);
+            }
+            else if ("array".equals(type) && nestedValue instanceof Collection<?> collection)
```

Comment:
I don't think we should ever write `"array"` according to the spec. From what I remember then an array of strings should have the type `"string"`, but I can't see this in the spec now.

Reply:
Found it here: https://hyland.atlassian.net/wiki/spaces/cin/pages/1817051141/ADR+0006+-+Data+format+for+Objects